From 4de76fa443ab107e4542bbdba242f0ed467cfecf Mon Sep 17 00:00:00 2001
From: law <law@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Tue, 13 May 2014 20:26:41 +0000
Subject: [PATCH] 	PR tree-optimization/60902
 	* tree-ssa-threadedge.c
 	(record_temporary_equivalences_from_stmts_at_dest): Only iterate
 	over real defs when invalidating outputs from statements that do not
 	produce useful outputs for threading.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_9-branch@210399 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/tree-ssa-threadedge.c |    2 +-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/gcc/tree-ssa-threadedge.c b/gcc/tree-ssa-threadedge.c
index 8a0103b..7621348 100644
--- a/gcc/tree-ssa-threadedge.c
+++ b/gcc/tree-ssa-threadedge.c
@@ -398,7 +398,7 @@ record_temporary_equivalences_from_stmts_at_dest (edge e,
 	  ssa_op_iter iter;
 
 	  if (backedge_seen)
-	    FOR_EACH_SSA_TREE_OPERAND (op, stmt, iter, SSA_OP_ALL_DEFS)
+	    FOR_EACH_SSA_TREE_OPERAND (op, stmt, iter, SSA_OP_DEF)
 	      {
 		/* This call only invalidates equivalences created by
 		   PHI nodes.  This is by design to keep the cost of
-- 
1.7.1

